### PR TITLE
Settings and jitsi features for video rooms

### PR DESCRIPTION
**Pull request by DashieTM**
image

Simple exposure for jitsi audio settings. --> opt-out
Dependent on: https://github.com/matrix-org/matrix-react-sdk/pull/8255
Feature request: https://github.com/vector-im/element-web/issues/21687

Signed-off-by: DashieTM <fabio.lenherr@gmail.com>

### **Notes:** 

- Added Buttons for jitsi audio configuration: echo cancellation, noise suppression, automatic gain, high pass filter

Type:[enhancement]

---
This PR currently has no changelog labels, so will not be included in changelogs.

A reviewer can add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is, or add `Type: [enhancement/defect/task]` to the description and I'll add them for you.

### CODE REVIEW

File: src/vector/jitsi/index.ts [hunk 1]
```diff
@@ -323,6 +324,11 @@ function joinConference() { // event handler bound in HTML
             SHOW_WATERMARK_FOR_GUESTS: false,
             MAIN_TOOLBAR_BUTTONS: [],
             VIDEO_LAYOUT_FIT: "height",
+            disableAEC: SettingsStore.getValue('Audio.disableAEC'),
+            disableNS: SettingsStore.getValue('Audio.disableNS'),
+            disableAP: SettingsStore.getValue('Audio.disableAP'),
+            disableAGC: SettingsStore.getValue('Audio.disableAGC'),
+            disableHPF: SettingsStore.getValue('Audio.disableHPF'),
```
Comment (turt2live) [inline src/vector/jitsi/index.ts:331]: These don't match the definitions in the react-sdk, and we really shouldn't be using the SettingsStore in the Jitsi widget.

We'd instead probably want to use the widget API to send settings information down into the widget.

**Review by turt2live — COMMENTED** [review]
like the react-sdk side, this is a light code review while product/design take a look.

File: src/vector/jitsi/index.ts (hunk 1 above)
Comment (robintown) [reply src/vector/jitsi/index.ts:331]: Or, alternatively, these could be included in the URL fragment of the Jitsi widget, like the `isVideoChannel` parameter

Comment (turt2live) [reply src/vector/jitsi/index.ts:331]: For audio settings like this I'd feel uncomfortable about having them so close to the URL processing, so would still recommend using the widget API as a cheap-ish way to do it.

I think we already have an MSC for generic config being sent to widgets anyhow.

Comment (DashieTM) [reply src/vector/jitsi/index.ts:331]: Oh god, I completely forgot about them when I renamed them for consistency. I am very sorry about that. 
And I will have a look into the API, the Settingstore was the first one that crossed my mind, but I will use something better if there is something.

**Pull request closed without merging**